src/main/java/me/chrr/scribble/mixin/BookEditScreenMixin.java

```diff
@@ -274,50 +315,92 @@ public void init(PlayerEntity player, ItemStack itemStack, Hand hand, CallbackIn
+        Optional.ofNullable(underlineButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(strikethroughButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(obfuscatedButton).ifPresent(button -> button.visible = !this.signing);
 
         for (ColorSwatchWidget swatch : colorSwatches) {
-            swatch.visible = !this.signing;
+            if (swatch != null) {
+                swatch.visible = !signing;
+            }
         }
 
-        this.deletePageButton.visible = !this.signing && this.richPages.size() > 1;
-        this.insertPageButton.visible = !this.signing;
+        Optional.ofNullable(deletePageButton).ifPresent(button -> button.visible = !signing && richPages.size() > 1);
+        Optional.ofNullable(insertPageButton).ifPresent(button ->
+                button.visible = !signing && richPages.size() < MAX_PAGES_NUMBER
+        );
 
-        this.saveBookButton.visible = !this.signing;
-        this.loadBookButton.visible = !this.signing;
+        Optional.ofNullable(saveBookButton).ifPresent(button -> button.visible = !signing);
+        Optional.ofNullable(loadBookButton).ifPresent(button -> button.visible = !signing);
     }
 
     @Unique
-    private void updateState(@Nullable Formatting color, Set<Formatting> modifiers) {
-        boldButton.toggled = modifiers.contains(Formatting.BOLD);
-        italicButton.toggled = modifiers.contains(Formatting.ITALIC);
-        underlineButton.toggled = modifiers.contains(Formatting.UNDERLINE);
-        strikethroughButton.toggled = modifiers.contains(Formatting.STRIKETHROUGH);
-        obfuscatedButton.toggled = modifiers.contains(Formatting.OBFUSCATED);
-        setSwatchColor(color);
+    private void invalidateFormattingButtons() {
+        Optional.ofNullable(boldButton).ifPresent(button -> button.toggled = activeModifiers.contains(Formatting.BOLD));
+        Optional.ofNullable(italicButton).ifPresent(button -> button.toggled = activeModifiers.contains(Formatting.ITALIC));
+        Optional.ofNullable(underlineButton).ifPresent(button -> button.toggled = activeModifiers.contains(Formatting.UNDERLINE));
+        Optional.ofNullable(strikethroughButton).ifPresent(button -> button.toggled = activeModifiers.contains(Formatting.STRIKETHROUGH));
+        Optional.ofNullable(obfuscatedButton).ifPresent(button -> button.toggled = activeModifiers.contains(Formatting.OBFUSCATED));
+        setSwatchColor(activeColor);
     }
 
     @Unique
     private void setSwatchColor(Formatting color) {
         for (ColorSwatchWidget swatch : colorSwatches) {
-            swatch.setToggled(swatch.getColor() == color);
+            if (swatch != null) {
+                swatch.setToggled(swatch.getColor() == color);
+            }
         }
     }
 
+    @Unique
+    private void onCursorFormattingChanged(@Nullable Formatting color, Set<Formatting> modifiers) {
+        this.activeColor = color != null ? color : DEFAULT_COLOR;
```

Comment:
The problem mentioned above is here, this should keep `null` if it happens. In that case, no swatch is selected, as there are multiple colors within the selection. If we make it black, you can't turn text with multiple colors back to black.